reactor-core/src/main/java/reactor/core/publisher/FluxScanSeed.java

```diff
@@ -187,12 +193,25 @@ public Object scanUnsafe(Attr key) {
 			return InnerOperator.super.scanUnsafe(key);
 		}
 
+		private boolean sendSeed() {
+			if (STATE.compareAndSet(this, 0, SENT_SEED)) {
+				serializedOnNext(value);
+				return true;
+			}
+			return false;
+		}
+
+		// concurrent calls to request at the start could yield concurrent calls to
+		// onNext. this will have low-to-no contention after the initial state has emitted
+		private synchronized void serializedOnNext(R value) {
```

Comment:
booger. thanks for being patient with me. (trying to avoid a queue! i had spiked out a single-element queue but could see how that could still overflow)